src/client/mod.rs

```diff
@@ -222,6 +234,14 @@ impl<'a> ClientBuilder<'a> {
         self
     }
 
+    /// Gets the cache settings. See [`Self::cache_settings`] for more info.
+    #[cfg(feature = "cache")]
+    pub fn get_cache_settings(&self) -> &CacheSettings {
+        // unwrap() is ok because cache_settings will only ever be None in the middle of being
+        // .await'ed
+        self.cache_settings.as_ref().unwrap()
+    }
```

Comment:
Clippy unfortunately disagrees here, so it must be informed that it is okay.
```suggestion
    /// Gets the cache settings. See [`Self::cache_settings`] for more info.
    #[cfg(feature = "cache")]
    #[allow(clippy::unwrap_used)]
    pub fn get_cache_settings(&self) -> &CacheSettings {
        // unwrap() is ok because cache_settings will only ever be None in the middle of being
        // .await'ed
        self.cache_settings.as_ref().unwrap()
    }
```